docs/whirlpool/docs/02-Architecture Overview/07-TokenExtensions Support.md

```diff
@@ -0,0 +1,161 @@
+
+#### `amount` and `otherAmountThreshold`
+The relationship between `amount`, `otherAmountThreshold` and `TransferFee` in each context.
+
+The relationship between amount, otherAmountThreshold and TransferFee in each context.
+
+ExactIn (`amount_specified_is_input = true`)
+- `amount`: transfer fee Included amount (will be sent from user's token account)
+- `otherAmountThreshold`: transfer fee Excluded amount (will be received at user's token account)
+
+ExactOut (`amount_specified_is_input = false`)
+- `amount`: transfer fee Excluded amount (will be received at user's token account)
+- `otherAmountThreshold`: transfer fee Included amount (will be sent from user's token account)
+
+The user can set conditions on the amount actually going out of and coming into the token account, regardless of the amount of fees.
+
+No additional parameters are added to limit the amount of fees.
+If the fee will be changed, it is forced to be activated two epochs after, so there is no possibility of the fee suddenly going up. The edge case is when the fee change is scheduled (new epoch is coming) in less than transaction life time. In this case, the UI may need to alert users, but in any case, transactions that exceed the outgoing and incoming amount thresholds will fail.
+
+#### `minA/B`, `maxA/B`
+Increase Liquidity: `maxA`, `maxB`
+transfer fee Included amount (will be sent from user's token account)
+
+Decrease Liquidity: `minA`, `minB`
+transfer fee Excluded amount (will be received at user's token account)
+
+### TransferHook
+In order to use TransferHook, the owner of WhirlpoolsConfig must issue a TokenBadge for that token.
+
+#### Use of remaining_accounts
+The account used for TransferHook is different for each program used by TransferHook. Therefore, they are received through `remaining_accounts`.
+
+To clarify the context of accounts passed as `remaining_accounts`, the v2 instructions receive `remaining_accounts_info` as data. It is used to classify the accounts contained in `remaining_accounts`. (e.g. The first 3 are for mintA's TransferHook, next 2 are for for mintB's TransferHook).
+
+#### Security notes
+- In solana, indirect re-entrance will be rejected. So transfer hook program cannot call Whirlpool program.
+- TransferHook program cannot modify transfer amount. It will be called AFTER token transfer processing.
+- Token-2022 program remove signer and writable flag of the source, destination and owner when calling the TransferHo[ok program even if they are passed as extra accounts. The source and destination are not updated and the owner's signature is not used unintentionally.
+
+#### TokenBadge request
+TransferHook extension is a relatively new feature in the ecosystem and it is still unknown how it will be used.
+
+It remains to be decided what kind of TransferHook tokens TokenBadge will be issued for, but it is important that they have the following characteristics
```

Comment:
> but it is important that they have the following characteristics

For who is this important? Might be worth to just call this best practices or something like that